coramin/third_party/minlplib_tools.py

```diff
@@ -0,0 +1,336 @@
+            logger.debug('excluding {case} due to ngennlcons'.format(case=case_name))
+            continue
+
+        njacobiannz = int(row[headings['njacobiannz']])
+        if njacobiannz < min_njacobiannz or njacobiannz > max_njacobiannz:
+            logger.debug('excluding {case} due to njacobiannz'.format(case=case_name))
+            continue
+
+        njacobiannlnz = int(row[headings['njacobiannlnz']])
+        if njacobiannlnz < min_njacobiannlnz or njacobiannlnz > max_njacobiannlnz:
+            logger.debug('excluding {case} due to njacobiannlnz'.format(case=case_name))
+            continue
+
+        nlaghessiannz = int(row[headings['nlaghessiannz']])
+        if nlaghessiannz < min_nlaghessiannz or nlaghessiannz > max_nlaghessiannz:
+            logger.debug('excluding {case} due to nlaghessiannz'.format(case=case_name))
+            continue
+
+        nlaghessiandiagnz = int(row[headings['nlaghessiandiagnz']])
+        if nlaghessiandiagnz < min_nlaghessiandiagnz or nlaghessiandiagnz > max_nlaghessiandiagnz:
+            logger.debug('excluding {case} due to nlaghessiandiagnz'.format(case=case_name))
+            continue
+
+        nsemi = int(row[headings['nsemi']])
+        if nsemi < min_nsemi or nsemi > max_nsemi:
+            logger.debug('excluding {case} due to nsemi'.format(case=case_name))
+            continue
+
+        nnlsemi = int(row[headings['nnlsemi']])
+        if nnlsemi < min_nnlsemi or nnlsemi > max_nnlsemi:
+            logger.debug('excluding {case} due to nnlsemi'.format(case=case_name))
+            continue
+
+        nsos1 = int(row[headings['nsos1']])
+        if nsos1 < min_nsos1 or nsos1 > max_nsos1:
+            logger.debug('excluding {case} due to nsos1'.format(case=case_name))
+            continue
+
+        nsos2 = int(row[headings['nsos2']])
+        if nsos2 < min_nsos2 or nsos2 > max_nsos2:
+            logger.debug('excluding {case} due to nsos2'.format(case=case_name))
+            continue
+
+        cases.append(case_name)
+
+    return cases
+
+
+def get_minlplib(download_dir=None, format='osil'):
+    """
+    Download MINLPLib
+
+    Parameters
+    ----------
+    download_dir: str
+        The directory in which to place the downloaded files. The default will be a 
+        current_working_directory/minlplib/file_format/.
+    format: str
+        The file format requested. Options are ams, gms, lp, mod, nl, osil, and pip
+    """
+    if download_dir is None:
+        download_dir = os.path.join(os.getcwd(), 'minlplib', format)
+    if not os.path.exists(download_dir):
+        os.makedirs(download_dir)
+
+    downloader = download.FileDownloader()
+    zip_filename = os.path.join(download_dir, 'minlplib_'+format+'.zip')
+    downloader.set_destination_filename(zip_filename)
+    downloader.get_binary_file('http://www.minlplib.org/minlplib_'+format+'.zip')
+    zipper = ZipFile(zip_filename, 'r')
+    zipper.extractall(download_dir)
+    os.remove(zip_filename)
```

Comment:
I would recommend using the downloader.get_zip_archive(): it has additional checks for potentially malicious file names in the zip archive.

Reply:
I figured something like this existed... Thanks.